vendors/nordic/boards/nrf52840-dk/ports/ota_pal_for_aws/ota_pal.c

```diff
@@ -0,0 +1,734 @@
+#include <stdio.h>
+#include <stdlib.h>
+
+/* FreeRTOS include. */
+#include "FreeRTOS.h"
+#include "ota.h"
+#include "ota_pal.h"
+#include "ota_interface_private.h"
+#include "ota_config.h"
+
+#include "asn1utility.h"
+#include "mbedtls/base64.h"
+#include "nrf_mbr.h"
+#include "nrf_sdm.h"
+#include "nrf_nvmc.h"
+#include "nrf_sdh_soc.h"
+#include "nrf_crypto.h"
+
+#include "event_groups.h"
+
+static const char codeSigningCertificatePEM[] = otapalconfigCODE_SIGNING_CERTIFICATE;
+
+/* Specify the OTA signature algorithm we support on this platform. */
+const char OTA_JsonFileSignatureKey[ OTA_FILE_SIG_KEY_STR_MAX_LENGTH ] = "sig-sha256-ecdsa";
+const char pcOTA_PAL_Magick[ otapalMAGICK_SIZE ] = "@AFRTOS";
+const size_t otapalFLASH_END = ( size_t ) &__FLASH_segment_end__;
+/* Tag by which the beginning of the ECDSA in the public key can be found */
+const char pcOTA_PAL_ASN_1_ECDSA_TAG[] = "\x06\x07\x2A\x86\x48\xCE\x3D\x02\x01\x06\x08\x2A\x86\x48\xCE\x3D\x03\x01\x07";
+
+const char pcOTA_PAL_CERT_BEGIN[] = "-----BEGIN CERTIFICATE-----";
+const char pcOTA_PAL_CERT_END[] = "-----END CERTIFICATE-----";
+static uint32_t pulSerializingArray[ otapalSERIALIZING_ARRAY_SIZE ];
+size_t ulPublicKeySize = 0;
+
+/* The static functions below (otaPal_CheckFileSignature and otaPal_ReadAndAssumeCertificate)
+ * are optionally implemented. If these functions are implemented then please set the following macros in
+ * aws_test_ota_config.h to 1:
+ * otatestpalCHECK_FILE_SIGNATURE_SUPPORTED
+ * otatestpalREAD_AND_ASSUME_CERTIFICATE_SUPPORTED
+ */
+
+/**
+ * @brief Verify the signature of the specified file.
+ *
+ * This function should be implemented if signature verification is not offloaded
+ * to non-volatile memory io functions.
+ *
+ * This function is called from otaPal_Close().
+ *
+ * @param[in] pFileContext OTA file context information.
+ *
+ * @return Below are the valid return values for this function.
+ * OtaPalSuccess if the signature verification passes.
+ * OtaPalSignatureCheckFailed if the signature verification fails.
+ * OtaPalBadSignerCert if the if the signature verification certificate cannot be read.
+ *
+ */
+static OtaPalMainStatus_t otaPal_CheckFileSignature( OtaFileContext_t * const pFileContext );
+
+/**
+ * @brief Read the specified signer certificate from the filesystem into a local buffer.
+ *
+ * The allocated memory returned becomes the property of the caller who is responsible for freeing it.
+ *
+ * This function is called from otaPal_CheckFileSignature(). It should be implemented if signature
+ * verification is not offloaded to non-volatile memory io function.
+ *
+ * @param[in] pucCertName The file path of the certificate file.
+ * @param[out] ulSignerCertSize The size of the certificate file read.
+ *
+ * @return A pointer to the signer certificate in the file system. NULL if the certificate cannot be read.
+ * This returned pointer is the responsibility of the caller; if the memory is allocated the caller must free it.
```

Comment:
Can you wrap these to be 80 characters long? Please apply this to the rest of the file as well.